app/assess/templates/macros/sub_criteria_navbar.html

```diff
@@ -10,9 +9,9 @@
     </ul>
 
 {% if g.user.highest_role in ["ASSESSOR","LEAD_ASSESSOR"] %}
-    {% if current_theme_id != "score" and sub_criteria.is_scored == True %}
+    {% if not is_score_page and sub_criteria.is_scored %}
     <ul class="govuk-list">
-        <li><a id="score-subcriteria-link" class="{{'govuk-link--no-underline govuk-list' if on_summary else 'govuk-link--no-visited-state'}}" href="{{ url_for('assess_bp.display_sub_criteria',application_id=application_id,sub_criteria_id=sub_criteria.id,theme_id='score') }}">Score the subcriteria</a></li>
+        <li><a id="score-subcriteria-link" class="{{'govuk-link--no-underline govuk-list' if is_score_page else 'govuk-link--no-visited-state'}}" href="{{ url_for('assess_bp.score',application_id=application_id,sub_criteria_id=sub_criteria.id) }}">Score the subcriteria</a></li>
```

Comment:
The `if is_score_page` is redundant as based on L12, we only render this block given it is `False`, so it will always go to the `else` statement

Reply:
@tferns Although it is defaulted in the macro, we can still override it by setting the argument which we do in https://github.com/communitiesuk/funding-service-design-assessment/blob/023370a1816aa6988f033ed79e8bf36c07a5bb45/app/assess/templates/score.html#L43

Reply:
Ah, I was referring to the encapsulating if statement, i.e:
https://github.com/communitiesuk/funding-service-design-assessment/blob/194bff396de7e89d1f088908c30154e01f3d21be/app/assess/templates/macros/sub_criteria_navbar.html#L12

This condition needs to be met for the `<ul...` to be rendered, so the nested if condition is never met:
https://github.com/communitiesuk/funding-service-design-assessment/blob/194bff396de7e89d1f088908c30154e01f3d21be/app/assess/templates/macros/sub_criteria_navbar.html#L14

i.e
```python3
is_not_true = False
if not is_not_true:
    if is_not_true: # will never be called
```

Reply:
Ah, I was referring to the encapsulating if statement:
https://github.com/communitiesuk/funding-service-design-assessment/blob/194bff396de7e89d1f088908c30154e01f3d21be/app/assess/templates/macros/sub_criteria_navbar.html#L12

This condition needs to be met for the `<ul...` to be rendered, so the nested if condition is never met:
https://github.com/communitiesuk/funding-service-design-assessment/blob/194bff396de7e89d1f088908c30154e01f3d21be/app/assess/templates/macros/sub_criteria_navbar.html#L14

i.e
```python3
is_not_true = False
if not is_not_true:
    if is_not_true: # will never be called
```

Reply:
Ah, I was referring to the encapsulating if statement:
https://github.com/communitiesuk/funding-service-design-assessment/blob/194bff396de7e89d1f088908c30154e01f3d21be/app/assess/templates/macros/sub_criteria_navbar.html#L12

This condition needs to be met for the `<ul...` to be rendered, so the nested if condition is never met:
https://github.com/communitiesuk/funding-service-design-assessment/blob/194bff396de7e89d1f088908c30154e01f3d21be/app/assess/templates/macros/sub_criteria_navbar.html#L14

i.e
```python3
is_not_true = False
if not is_not_true:
    if is_not_true: # will never be called
```

Reply:
```suggestion
        <li><a id="score-subcriteria-link" class="govuk-link--no-visited-state" href="{{ url_for('assess_bp.score',application_id=application_id,sub_criteria_id=sub_criteria.id) }}">Score the subcriteria</a></li>
```

Reply:
@tferns ah yes, I agree the code is unreachable but I believe the same bug is present as-is in `main` e.g. https://assessment.uat.gids.dev/assess/application_id/0b3e5a51-e9a1-4000-9d27-79e111edf737/sub_criteria_id/benefits?theme_id=score so would it be OK to fix this in a separate PR, rather than changing the observed behaviour in this one?